docs: add Rule Extensions GA documentation (#1446)
snyk/user-docs: scan-fix-and-prevent/scan-with-snyk/snyk-code/rule-extensions/custom-sanitizers.md

# Custom sanitizers

Taint flow analysis can miss your application's unique security controls. Defining custom sanitizers helps you get more accurate results by teaching the scanner about your specific data cleaning and validation methods, reducing false positives.

> Here's when to use them:

Reviewer: ```suggestion
Use custom sanitizers in the following cases:
```

        if (data == null) return false;
        return data.toLowerCase().trim().startsWith("javascript:");
    }
}
```
{% endcode %}

It is used in this HttpServlet:

```java
import com.company.utils.SecurityUtils;
import javax.servlet.http.HttpServlet;
import javax.servlet.http.HttpServletRequest;
import javax.servlet.http.HttpServletResponse;
import java.io.IOException;

public class RedirectServlet extends HttpServlet {
    protected void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
        var redirectUrl = request.getParameter("url");

        if (!SecurityUtils.containsJavascriptProtocol(redirectUrl)) {
            response.sendRedirect(redirectUrl);
        } else {
            response.sendError(400, "Invalid redirect URL detected.");
        }
    }
}
```
{% endtab %}
{% endtabs %}

### Any Usage

_API `extension_type`: `any_usage_sanitizer`_

Arguments to these sanitizers are considered sanitized after the execution of the sanitizer. This form of sanitizer is either expected to mutate data that a reference is passing, or throw exceptions if non-sanitized data is passed.

> If it was used as the wrong kind for the sanitization, it could lead to false negatives and hide incorrect usage of sanitizer functions in the code.

Reviewer: ```suggestion
Using the wrong sanitizer type can cause false negatives and hide incorrect usage of sanitizer functions in your code.
```
